src/components/text-highlighter.vue

```diff
@@ -0,0 +1,23 @@
+<template>
+  <Highlighter
+    v-if="enable"
+    highlight-class-name="font-bold underline bg-highlight-color"
+    :text-to-highlight="textToHighlight"
+    :search-words="modelValue?.split(' ') || []"
+  />
+  <span v-else>{{ textToHighlight }}</span>
+</template>
+
+<script lang="ts" setup>
+import Highlighter from 'vue-highlight-words';
+
+interface IProps {
+  modelValue?: null | string;
+  textToHighlight: string;
+  enable: boolean;
+}
+
+defineProps<IProps>();
+</script>
+
+<style lang="postcss" scoped></style>
```

Comment:
```suggestion
<template>
  <Highlighter
    v-if="enable"
    highlight-class-name="font-bold underline bg-highlight-color"
    :text-to-highlight="textToHighlight"
    :search-words="searchWords"
  />
  <span v-else>{{ textToHighlight }}</span>
</template>

<script lang="ts" setup>
import Highlighter from 'vue-highlight-words';

interface IProps {
  modelValue?: null | string;
  textToHighlight: string;
  enable: boolean;
}

const props = defineProps<IProps>();

const searchWords = computed(() => props.modelValue?.split(' ') || []);
</script>
```

Make use of `computed` so this computation gets cached and won't have to do this logic whenever, for whatever reason, upstream components forces this component to re-render the template.